p11-kit/modules.c

```diff
@@ -199,7 +199,7 @@ const char *p11_config_user_modules = P11_USER_CONFIG_MODULES;
  * of p11-kit usage.
  *
  * When the value %NULL is provided for any of the locations,
- * it will not be updated.
+ * the default location will be used.
```

Comment:
Nice catch, that could affect gnutls' usage of it. It is used in the test suite which calls:
`p11_kit_override_system_files(NULL, NULL, mod, mod, NULL);` and it expects that NULL will not load anything. Not sure what the default values are set for the parameters that are set to null. What if the value ((void*)(-1)) is used for new semantics?

Reply:
Well, there seems to be a confusion here.  Previously, `NULL` merely meant not to change the previous setting (as documented).  That is, unless you call `p11_kit_override_system_files()` multiple times and the preceding call had set a non-existing location, you can't expect it will not load anything.

Also I don't quite like adding a special value; you can simply pass a non-existing path e.g., `/nonexistent`.

Reply:
Changing the gnutls test it is easy, so we can change it if the behavior changes. Should we use "/nonexistent" instead?

Reply:
The behavior change shouldn't affect the gnutls test actually. Given that there is no special handling of NULL set to those variables, you need to use "/nonexistent" if you want to disable loading.